lib/generateMarkdown.js

```diff
@@ -103,14 +103,26 @@ function getRecursiveTOC(orderedTypes, parentType, depth) {
     for (var i = 0; i < orderedTypes[parentType].children.length; i++) {
         var currentType = orderedTypes[parentType].children[i];
         if (orderedTypes[currentType].parents.length == 1) {
-            md += style.bulletItem(style.linkType(style.typeValue(currentType), currentType, enums.autoLinkOption.codeQuoteOnly), depth);
+            md += style.bulletItem(getTOCLink(currentType.replace(parentType + " ", ""), currentType), depth);
             md += getRecursiveTOC(orderedTypes, currentType, depth + 1);
         }
     }
     return md;
 }
 
 /**
+* @function getTOCLink
+* @param  {string} displayString The text to display in the link.
+* @param  {string} type          The string to link to.
+* @return {string} The markdown for a link with displayString text targeted at type.
+*/
+function getTOCLink(displayString, type)
+{
+    var typeLink = '#reference-' + type.toLowerCase().replace(/ /g, '-');
+    return "[" + style.typeValue(displayString) + '](' + typeLink + ")";
```

Comment:
We already have a method that knows how to make MD links: `style.getLinkMarkdown`.  Please use that.